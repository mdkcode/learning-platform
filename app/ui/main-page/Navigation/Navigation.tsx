"use client";
import Link from "next/link";
import clsx from "clsx";
import { ButtonType } from "@/app/ui/components/Button/utils";
import { Routes } from "@/app/routes/routes";
import Button from "@/app/ui/components/Button/Button";
import { handleSignIn } from "@/app/api/auth/[...nextauth]/auth";

export interface NavigationProps {
  className?: string;
  buttonClassName?: string;
  buttonType?: ButtonType;
}

const Navigation = ({
  className,
  buttonClassName,
  buttonType,
}: NavigationProps) => {
  return (
    <div
      className={clsx(
        "flex flex-row items-center justify-between py-4",
        className
      )}
    >
      <Link href={Routes.HOME}>
        <h4 className="text-2xl font-extrabold">SkillEd</h4>
      </Link>
      <Button
        label="Get started"
        buttonType={ButtonType.NAVIGATION}
        className={buttonClassName}
        onClick={handleSignIn}
      />
    </div>
  );
};

export default Navigation;
