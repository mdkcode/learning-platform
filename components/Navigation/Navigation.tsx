import Link from "next/link";
import NavButton from "../Button/NavButton";
import clsx from "clsx";
import { ButtonType } from "../Button/utils";

export interface NavigationProps {
  className?: string;
  buttonClassName?: string;
  buttonType?: ButtonType;
}

const Navigation = ({
  className,
  buttonClassName,
  buttonType,
}: NavigationProps) => {
  return (
    <div
      className={clsx(
        "flex flex-row items-center justify-between py-4",
        className
      )}
    >
      <Link href="/">
        <h4 className="text-2xl font-extrabold">SkillEd</h4>
      </Link>
      <NavButton className={buttonClassName} buttonType={buttonType} />
    </div>
  );
};

export default Navigation;
