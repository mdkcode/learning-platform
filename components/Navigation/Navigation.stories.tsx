import React from "react";
import { Meta, StoryFn } from "@storybook/react";
import Navigation, { NavigationProps } from "./Navigation";
import { ButtonType } from "components/Button/utils";

export default {
  title: "components/Navigation",
  component: Navigation,
  parameters: {
    nextjs: {
      appDirectory: true,
    },
  },
} as Meta;

const Template: StoryFn<NavigationProps> = (args) => <Navigation {...args} />;

export const Header = Template.bind({});
Header.args = {
  buttonType: ButtonType.PRIMARY,
};

export const Footer = Template.bind({});
Footer.args = {
  buttonType: ButtonType.NAVIGATION,
};
