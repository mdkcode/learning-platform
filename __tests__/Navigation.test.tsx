import { render } from "@testing-library/react";
import { describe, it, expect, vi } from "vitest";
import Navigation from "@/app/ui/main-page/Navigation/Navigation";

vi.mock("@/app/api/auth/[...nextauth]/auth", () => ({
  handleSignIn: vi.fn(),
}));

describe("Navigation", () => {
  it("matches snapshot", () => {
    const { container } = render(<Navigation />);
    expect(container).toMatchSnapshot();
  });

  it("matches snapshot with custom button class", () => {
    const { container } = render(
      <Navigation buttonClassName="bg-blue-500 text-white" />
    );
    expect(container).toMatchSnapshot();
  });
});
