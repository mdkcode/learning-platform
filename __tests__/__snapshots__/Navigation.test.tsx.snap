// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Navigation > matches snapshot 1`] = `
<div>
  <div
    class="flex flex-row items-center justify-between py-4"
  >
    <a
      href="/"
    >
      <h4
        class="text-2xl font-extrabold"
      >
        SkillEd
      </h4>
    </a>
    <button
      class="px-7 py-3 text-white font-semibold rounded-lg bg-gradient-to-r from-orange-400 to-pink-500 hover:bg-gradient-to-r hover:from-orange-500 hover:to-pink-600"
    >
      Get started
    </button>
  </div>
</div>
`;

exports[`Navigation > matches snapshot with custom button class 1`] = `
<div>
  <div
    class="flex flex-row items-center justify-between py-4"
  >
    <a
      href="/"
    >
      <h4
        class="text-2xl font-extrabold"
      >
        SkillEd
      </h4>
    </a>
    <button
      class="px-7 py-3 text-white font-semibold rounded-lg bg-gradient-to-r from-orange-400 to-pink-500 hover:bg-gradient-to-r hover:from-orange-500 hover:to-pink-600 bg-blue-500 text-white"
    >
      Get started
    </button>
  </div>
</div>
`;
